src/Acuminator/Acuminator.Analyzers/StaticAnalysis/PXGraphSavingChangesDuringInitialization/PXGraphSavingChangesDuringInitializationAnalyzer.cs

```diff
@@ -0,0 +1,75 @@
+﻿using Acuminator.Analyzers.StaticAnalysis.PXGraph;
+using Acuminator.Analyzers.StaticAnalysis.SavingChanges;
+using Acuminator.Utilities.Roslyn;
+using Acuminator.Utilities.Roslyn.Semantic.PXGraph;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using System.Collections.Immutable;
+
+namespace Acuminator.Analyzers.StaticAnalysis.PXGraphSavingChangesDuringInitialization
+{
+    public class PXGraphSavingChangesDuringInitializationAnalyzer : IPXGraphAnalyzer
+    {
+        public ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics =>
+            ImmutableArray.Create
+            (
+                Descriptors.PX1058_PXGraphSavingChangesDuringInitialization
+            );
+
+        public void Analyze(SymbolAnalysisContext context, PXContext pxContext, PXGraphSemanticModel pxGraph)
+        {
+            context.CancellationToken.ThrowIfCancellationRequested();
+
+            foreach (GraphInitializerInfo initializer in pxGraph.Initializers)
+            {
+                context.CancellationToken.ThrowIfCancellationRequested();
+                CheckSavingChanges(context, pxContext, initializer);
+            }
+        }
+
+        private void CheckSavingChanges(SymbolAnalysisContext context, PXContext pxContext, GraphInitializerInfo initializer)
+        {
+            context.CancellationToken.ThrowIfCancellationRequested();
+
+            if (initializer.Node == null)
+                return;
+
+            SaveChangesWalker walker = new SaveChangesWalker(context, pxContext);
+
+            walker.Visit(initializer.Node);
+        }
+
+        private class SaveChangesWalker : CSharpSyntaxWalker
```

Comment:
@metallmonster @dropsonic I suppose here the recursive analysis is also expected

Reply:
@SENya1990 ok